packages/orchestration/src/exos/portfolio-holder-kit.js

```diff
@@ -0,0 +1,176 @@
+ * @import {MapStore} from '@agoric/store';
+ * @import {VowTools} from '@agoric/vow';
+ * @import {ResolvedPublicTopic} from '@agoric/zoe/src/contractSupport/topics.js';
+ * @import {Zone} from '@agoric/zone';
+ * @import {OrchestrationAccount, OrchestrationAccountI} from '@agoric/orchestration';
+ */
+
+/**
+ * @typedef {{
+ *   accounts: MapStore<string, OrchestrationAccount<any>>;
+ *   publicTopics: MapStore<string, ResolvedPublicTopic<unknown>>;
+ * }} PortfolioHolderState
+ */
+
+const ChainNameShape = M.string();
+
+const AccountEntriesShape = M.arrayOf([
+  M.string(),
+  M.remotable('OrchestrationAccount'),
+]);
+const PublicTopicEntriesShape = M.arrayOf([M.string(), PublicTopicShape]);
+
+/**
+ * Kit that holds several OrchestrationAccountKits and returns a invitation
+ * makers.
+ *
+ * @param {Zone} zone
+ * @param {VowTools} vowTools
+ */
+const preparePortfolioHolderKit = (zone, { asVow, when }) => {
+  return zone.exoClassKit(
+    'PortfolioHolderKit',
+    {
+      invitationMakers: M.interface('InvitationMakers', {
+        Action: M.call(ChainNameShape, M.string(), M.arrayOf(M.any())).returns(
+          M.promise(),
+        ),
+      }),
+      holder: M.interface('Holder', {
+        asContinuingOffer: M.call().returns(
+          Vow$({
+            publicSubscribers: TopicsRecordShape,
+            invitationMakers: M.any(),
+          }),
+        ),
+        getPublicTopics: M.call().returns(Vow$(TopicsRecordShape)),
+        getAccount: M.call(ChainNameShape).returns(Vow$(M.remotable())),
+        addAccount: M.call(
+          ChainNameShape,
+          M.remotable(),
+          PublicTopicShape,
+        ).returns(VowShape),
+      }),
+    },
+    /**
+     * @param {Iterable<[string, OrchestrationAccount<any>]>} accountEntries
+     * @param {Iterable<[string, ResolvedPublicTopic<unknown>]>} publicTopicEntries
+     */
+    (accountEntries, publicTopicEntries) => {
+      mustMatch(accountEntries, AccountEntriesShape, 'must provide accounts');
+      mustMatch(
+        publicTopicEntries,
+        PublicTopicEntriesShape,
+        'must provide public topics',
+      );
+      const accounts = harden(
+        makeScalarBigMapStore('accounts', { durable: true }),
+      );
+      const publicTopics = harden(
+        makeScalarBigMapStore('publicTopics', { durable: true }),
+      );
+      accounts.addAll(accountEntries);
+      publicTopics.addAll(publicTopicEntries);
+      return /** @type {PortfolioHolderState} */ (
+        harden({
+          accounts,
+          publicTopics,
+        })
+      );
+    },
+    {
+      invitationMakers: {
+        /**
+         * @template {unknown[]} IA
+         * @param {string} chainName key where the account is stored
+         * @param {string} action invitation maker name, e.g. 'Delegate'
+         * @param {IA} invitationArgs
+         * @returns {Promise<Invitation<unknown, IA>>}
+         */
+        Action(chainName, action, invitationArgs) {
```

Comment:
"Action" is very vague to me. this is making an invitation, right?

consider `MakeInvitation` or `ContinueInvitation`